controllers/machine_controller_phases.go

```diff
@@ -97,9 +96,18 @@ func (r *MachineReconciler) reconcileExternal(ctx context.Context, cluster *clus
 	obj, err := external.Get(ctx, r.Client, ref, m.Namespace)
 	if err != nil {
 		if apierrors.IsNotFound(errors.Cause(err)) {
-			return external.ReconcileOutput{}, errors.Wrapf(&capierrors.RequeueAfterError{RequeueAfter: externalReadyWait},
-				"could not find %v %q for Machine %q in namespace %q, requeuing",
-				ref.GroupVersionKind(), ref.Name, m.Name, m.Namespace)
+			// if the reference missing is the infrastructureRef and the machine was up and running, this is an error.
+			if *ref == m.Spec.InfrastructureRef && m.Status.InfrastructureReady {
+				err := errors.Errorf("could not find %v %q for Machine %q in namespace %q, requeuing", ref.GroupVersionKind(), ref.Name, m.Name, m.Namespace)
```

Comment:
This isn't requeue-ing with a fixed timestamp like before?

Reply:
Reverted to having this logic in reconcileInfrastructure